tools/ports/sdl2.py

```diff
@@ -9,16 +9,18 @@
 HASH = 'b178bdc8f7c40271e09a72f639649d1d61953dda4dc12b77437259667b63b961fd3b2c67b0de6fdc5f9f9c80c49bfafd164e4c13715bc1056e550acc8bad5a3c'
 SUBDIR = 'SDL-' + TAG
 
-variants = {'sdl2-mt': {'USE_PTHREADS': 1}}
-
+variants = {'sdl2-mt': {'USE_PTHREADS': 1}, 'sdl2-ww': {'WASM_WORKERS': 1}}
 
 def needed(settings):
   return settings.USE_SDL == 2
 
-
 def get_lib_name(settings):
-  return 'libSDL2' + ('-mt' if settings.USE_PTHREADS else '') + '.a'
-
+  name = 'libSDL2'
+  if settings.USE_PTHREADS:
+    name += '-mt'
+  elif settings.WASM_WORKERS:
+    name += '-ww'
+  return name + '.a'
```

Comment:
For both this library and libpng I suggest you start by simply using the existing `-mt` variant for `WASM_WORKERS`.   Only if that doesn't work for some reason should we consider a new variant.

Reply:
I removed the libpng-ww variants. Just use libpng-mt variants.

For sdl2-ww variants, I changed the code according to the comment here:
https://github.com/emscripten-core/emscripten/issues/18162#issuecomment-1308525594

The main difference is the thread_backend used by the SDL2.
PThread: use pthread
WASM worker: use generic

But I can't find the "wasm-worker" thread mentioned in the comments, so I just change it to use the generic

![image](https://user-images.githubusercontent.com/1687930/208614472-36d2d302-effe-4642-8f4a-520d8daafa61.png)

Reply:
what is in the implementation of "generic" threads?  what will happy if somebody tried to use SDL locking or threading primitives in the wasm workers build?  will it work on not work?

Reply:
I don't think the SDL locking or threading primitive will work. The generic implementation is just a stub https://github.com/libsdl-org/SDL/blob/main/src/thread/generic/SDL_systhread.c . The fix on libpng and sdl2_image is more important to my work, so I believe it is better to revert the sdl2.py at this moment?

Reply:
Sounds good.  Keeping this change as small as possible makes sense to me.

Reply:
@sbc100 I revert the change of SDL2, also replied to the comment about sdl2_image above. Thanks